torch/_refs/nn/functional/__init__.py

```diff
@@ -0,0 +1,48 @@
+import torch
+
+import torch._prims.utils as utils
+from torch._prims.utils import TensorLikeType, TensorLike, NumberType, Number
+import torch._refs as refs
+
+from typing import Optional
+
+all = [
+    "elu",
+]
+
+# elu is implemented specially because it has an alpha argument
+def elu(a: TensorLikeType, alpha: Optional[NumberType] = None, inplace: bool = False):
+    """
+    Reference implementation of torch.nn.functional.elu
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    # Type checks
+    assert isinstance(a, TensorLike)
+    assert alpha is None or isinstance(alpha, Number)
+
+    computation_dtype, result_dtype = refs._elementwise_dtypes(
+        a, type_promotion=refs.ELEMENTWISE_TYPE_PROMOTION_KIND.OP_MATH
+    )
+
+    (a,) = refs._convert_dtype(a, dtype=computation_dtype)
+
+    if alpha is not None:
+        alpha_promotion_type = utils.dtype_to_type(computation_dtype)
+        assert utils.is_lesser_type(type(alpha), alpha_promotion_type)
+        alpha = alpha_promotion_type(alpha)
+
+    # Performs computation
+    x = refs.maximum(0, a)
+
+    y = refs.sub(refs.exp(a), 1)
```

Comment:
We could do something where it looks like you're calling `torch.add` and then we're always remapping it to `refs.sub` unless you take over the op, but that seems circuitious

Reply:
IMO, it makes sense to provide that configuration option. For example, if somebody wants to decompose elu but they don't necessarily want explicit broadcasts in the graph.

Reply:
I don't think there should be a config knob. We should just call the torch functions.

Reply:
> I don't think there should be a config knob. We should just call the torch functions.

That's what I mean by a "config knob" :P Like, calling the torch functions allows a tracer to choose to interject at `__torch_function__`.

Reply:
OK I think I understand some of our confusion yesterday now.

@Chillee when you "take over sub" whenever there is a "refs.sub" call here you will not trace through it. Instead you see a torch/aten sub call in the final trace. Only when we choose to decompose a torch/aten operation into prims do the references come into it, and those references are either resolved to torch/aten ops (because a trace executor wants them just as they are) or they're resolved to prims or a mixture of the two. (In our current plan) you never see a trace with refs.foo in it (for all foo).